controllers/openstacknet_controller.go

```diff
@@ -197,11 +220,97 @@ func (r *OpenStackNetReconciler) Reconcile(ctx context.Context, req ctrl.Request
+		// NNCP resource struct object in order to examine its status, but the NNCP was created earlier via a server-side
+		// apply.  Thus we don't already have the struct object in memory, and must acquire it by asking for a list of
+		// NNCPs filtered by our label selector (which should return just the one resource we seek)
+		// err = r.Client.Get(context.TODO(), types.NamespacedName{Name: ncp.Name, Namespace: instance.Namespace}, nncp)
+		err = r.Client.List(context.TODO(), nncpList, &client.ListOptions{
+			LabelSelector: labels.SelectorFromSet(common.GetLabels(instance, openstacknet.AppLabel, map[string]string{})),
+		})
+
+		if err != nil {
+			instance.Status.CurrentState = ospdirectorv1beta1.NetError
+			_ = r.setStatus(instance, oldStatus, err.Error())
+			return ctrl.Result{}, err
+		}
+
+		if len(nncpList.Items) < 1 {
+			msg := fmt.Sprintf("OpenStackNet %s associated NodeNetworkConfigurationPolicy is still initializing, requeuing...", instance.Name)
+			r.Log.Info(msg)
+			instance.Status.CurrentState = ospdirectorv1beta1.NetInitializing
+			err = r.setStatus(instance, oldStatus, msg)
+			return ctrl.Result{RequeueAfter: time.Second * 5}, err
+		}
+
+		nncp := nncpList.Items[0]
+
+		netState = ospdirectorv1beta1.NetConfiguring
+		netStateMsg = fmt.Sprintf("OpenStackNet %s is configuring targeted node(s)", instance.Name)
+
+		for _, condition := range nncp.Status.Conditions {
+			if condition.Status == corev1.ConditionTrue {
+				if condition.Type == nmstateapi.NodeNetworkConfigurationPolicyConditionAvailable {
+					netState = ospdirectorv1beta1.NetConfigured
+					netStateMsg = fmt.Sprintf("OpenStackNet %s has been successfully configured on targeted node(s)", instance.Name)
+					break
+				} else if condition.Type == nmstateapi.NodeNetworkConfigurationPolicyConditionDegraded {
+					instance.Status.CurrentState = ospdirectorv1beta1.NetError
+					_ = r.setStatus(instance, oldStatus, err.Error())
+					return ctrl.Result{}, err
+				}
+			}
+		}
 	}
 
-	return ctrl.Result{}, nil
+	// Count IP assignments and update status
+	totalIps := 0
+	activeIps := 0
+
+	for _, roleIps := range instance.Status.RoleReservations {
+		for _, roleIP := range roleIps.Reservations {
+			totalIps++
+			if !roleIP.Deleted {
+				activeIps++
+			}
+		}
+	}
+
+	instance.Status.ReservedIPCount = totalIps
+	instance.Status.ActiveIPCount = activeIps
+	instance.Status.CurrentState = netState
+
+	err = r.setStatus(instance, oldStatus, netStateMsg)
+	return ctrl.Result{}, err
+}
+
+func (r *OpenStackNetReconciler) setStatus(instance *ospdirectorv1beta1.OpenStackNet, oldStatus *ospdirectorv1beta1.OpenStackNetStatus, msg string) error {
+
+	if msg != "" {
+		r.Log.Info(msg)
+	}
+
+	if oldStatus.CurrentState != instance.Status.CurrentState || oldStatus.ActiveIPCount != instance.Status.ActiveIPCount || oldStatus.ReservedIPCount != instance.Status.ReservedIPCount {
```

Comment:
should we use DeepEqual instead of != to be in sync with osbms controller?

Reply:
Sounds good.  Will change that.